include/LightGBM/dataset.h

```diff
@@ -947,6 +956,7 @@ class Dataset {
   std::vector<int> feature_need_push_zeros_;
   std::vector<std::vector<float>> raw_data_;
   bool wait_for_manual_finish_;
+  int omp_max_threads_ = 16;  // TODO(svotaw) decide before checkin
```

Comment:
Can we use the default value to -1? which means it uses the omp_num_threads?

Reply:
no, because that's the main issue.  omp_num_threads can be different on different calling threads, so we need a constant.